### PR TITLE
add python example and related api

### PR DESCRIPTION
**Pull request by wzhongyuan**
Add python example for image classification

### CODE REVIEW

**Comment by wzhongyuan** [comment]
@SeaOfOcean  Added README.MD and also updated bigdl dependency to snapshot as per your requirement

**Comment by wzhongyuan** [comment]
@SeaOfOcean  updated as per discussion

File: models/src/main/python/bigdl/zoo/example/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
+## Image Classification Python example
+This example illustrates how to load and predict for Imagenet models with given image path
+### Run steps
+1. Build 
+
+In order to run this example locally, we should build the project with `all-in-one` profile
+
+mvn clean package -P all-in-one
+
```
Comment (SeaOfOcean) [inline models/src/main/python/bigdl/zoo/example/README.md]: ./build.sh all-in-one?

File: models/src/main/python/bigdl/zoo/example/README.md [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+2. Setup environment
+
+Add model jar to `BIGDL_CLASSPATH` and python zip file to `PYTHONPATH`
+
+export DL_PYTHON_HOME=${path_to_analytics-zoo}/models/target/models-0.1-SNAPSHOT-python-api.zip
```
Comment (SeaOfOcean) [inline models/src/main/python/bigdl/zoo/example/README.md]: remove this?

Comment (wzhongyuan) [reply models/src/main/python/bigdl/zoo/example/README.md]: Refined

File: models/src/main/python/bigdl/zoo/example/README.md (hunk 1 above)
Comment (wzhongyuan) [reply models/src/main/python/bigdl/zoo/example/README.md]: @SeaOfOcean  Thanks, refined and verified

File: models/src/main/python/bigdl/zoo/example/README.md [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+Add model jar to `BIGDL_CLASSPATH` and python zip file to `PYTHONPATH`
+
+export DL_PYTHON_HOME=${path_to_analytics-zoo}/models/target/bigdl-models-0.1-SNAPSHOT-python-api.zip
+
+export PYTHONPATH=$PYTHONPATH:${path_to_bigdl}/dist/conf/spark-bigdl.conf:$DL_PYTHON_HOME
+
+export BIGDL_CLASSPATH=$(find ${path_to_bigdl}/dist/lib -name "*with-dependencies.jar" | head -n 1)
+export BIGDL_CLASSPATH=$BIGDL_CLASSPATH:${path_to_analytics-zoo}/models/target/models-0.1-SNAPSHOT-jar-with-dependencies.jar
+
```
Comment (SeaOfOcean) [inline models/src/main/python/bigdl/zoo/example/README.md]: The above export  BIGDL_CLASSPATH is duplicate?

Comment (wzhongyuan) [reply models/src/main/python/bigdl/zoo/example/README.md]: One is for bigdl and one is for models

Comment (SeaOfOcean) [reply models/src/main/python/bigdl/zoo/example/README.md]: bigdl is already packaged in models, no need for bigdl

Comment (wzhongyuan) [reply models/src/main/python/bigdl/zoo/example/README.md]: @SeaOfOcean   ah yes in `all-in-one` , thanks. refined

File: models/src/main/python/bigdl/zoo/example/ImageClassificationPredict.py [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+parser.add_argument('modelPath', help="Path where the model is stored")
+parser.add_argument('imgPath', help="Path where the images are stored")
+parser.add_argument('topN', type= int,  help="Number to specify top results")
+
+def predict(modelPath, imgPath, topN):
+    print "ImageClassification prediction"
+    print ("Model Path %s" % modelPath)
+    print ("Image Path %s" % imgPath)
+    print  ("Top N : %d" % topN)
+    labelMap = read_imagenet_label_map()
```
Comment (jason-dai) [inline models/src/main/python/bigdl/zoo/example/ImageClassificationPredict.py]: we should get labmap from predictor, just as in the scala examples

Comment (wzhongyuan) [reply models/src/main/python/bigdl/zoo/example/ImageClassificationPredict.py]: @jason-dai  ok, i'll make a minor change for it

Comment (wzhongyuan) [reply models/src/main/python/bigdl/zoo/example/ImageClassificationPredict.py]: @jason-dai   Updated in https://github.com/intel-analytics/analytics-zoo/pull/105

**Pull request merged**